### ISSUE TITLE
Security Issue: Failing npm audit

### ISSUE THREAD

**Issue by MattRiddell** [Unitech/pm2#4464]
##  What's going wrong?

`npm audit` fails:

```shell script
                       === npm audit security report ===

┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Machine-In-The-Middle                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ https-proxy-agent                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @pm2/io                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @pm2/io > @pm2/agent-node > proxy-agent > https-proxy-agent  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1184                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Machine-In-The-Middle                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ https-proxy-agent                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @pm2/io                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @pm2/io > @pm2/agent-node > proxy-agent > pac-proxy-agent >  │
│               │ https-proxy-agent                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1184                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

## How could we reproduce this issue?

run `npm audit`

pm2 --version: 3.5.1
node --version: v10.16.3
npm --version: 6.12.0

**Comment by axago** [comment]
![_audit_](https://user-images.githubusercontent.com/44166656/67162158-6265fa00-f359-11e9-8e94-013663e8989c.png)

**Comment by Unitech** [comment]
Fixed via publish of node-https-agent on 3.1.1
https://github.com/TooTallNate/node-proxy-agent

**Comment by axago** [comment]
Still getting the security warning, is this normal ?
![hg](https://user-images.githubusercontent.com/44166656/67531558-0d104c80-f6bb-11e9-92e1-f69ec15dacf1.png)
